pkg/network/transport/object/grpc/replication.go

```diff
@@ -190,3 +205,41 @@ func objectFromMessage(gMsg *objectGRPC.Object) (*object.Object, error) {
 
 	return object.NewFromV2(&msg), nil
 }
+
+func (s *Server) metaInfoSignature(o object.Object) ([]byte, error) {
+	var deleted []oid.ID
+	var locked []oid.ID
+	switch o.Type() {
+	case object.TypeTombstone:
+		var t object.Tombstone
+		err := t.Unmarshal(o.Payload())
+		if err != nil {
+			return nil, fmt.Errorf("reading tombstoned objects: %w", err)
+		}
+
+		deleted = t.Members()
+	case object.TypeLock:
+		var l object.Lock
+		err := l.Unmarshal(o.Payload())
+		if err != nil {
+			return nil, fmt.Errorf("reading locked objects: %w", err)
+		}
+
+		locked = make([]oid.ID, l.NumberOfMembers())
+		l.ReadMembers(locked)
+	default:
+	}
+
+	metaInfo := objectcore.MetaInfo(o.GetContainerID(), o.GetID(), o.PayloadSize(), deleted, locked, o.CreationEpoch())
+
+	var sig neofscrypto.Signature
+	err := sig.Calculate(s.signer, metaInfo)
```

Comment:
this differs from the original protocol [definition](https://github.com/nspcc-dev/neofs-api/blob/01229b49e9165e693a48e2b96ccad528df34b55e/object/service.proto#L725-L728)

do I understand correctly that we decided to limit the amount of information being signed, and this approach will eventually be backported into the protocol?

Reply:
yes, things have changed with time (or i did not get them from the beginning). we have discussed it with @roman-khimov, a custom message should be provided (see the 2. in the [comment](https://github.com/nspcc-dev/neofs-node/issues/2876#issuecomment-2319693139) mostly)

> and this approach will eventually be "backported" into the protocol?

yes, that is right. not release was done since so it should not be a problem